vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -36,11 +36,15 @@
 }
 
 
-def _filter_between(series: pd.Series, value: List[float]) -> pd.Series:
+def _filter_between(series: pd.Series, value: Union[List[float], List[str]]) -> pd.Series:
+    if is_datetime64_any_dtype(series):
+        value = pd.to_datetime(value)
```

Comment:
```suggestion
        # Each value will always have time 00:00:00. In order for the filter to include all times during
        # the end date value[1] we need to remove the time part of every value in series so that it's 00:00:00.
        value = pd.to_datetime(value)
        series = pd.to_datetime(series.dt.date)
```

This is my suggested solution here so that the same `return` line below works for both dates and numbers.